### PR TITLE
Create contact-us form for HPC

### PR DESCRIPTION
**Pull request by carkod**
## Done

Dynamic contact us form for HPC

## QA

Compare with [copy docs](https://docs.google.com/document/d/1Tdu0Z60hSE8BSMMYAShfF5KFCJ_tE11eRp31oryk5xg/edit#)

Check that text changes correctly according to different product names:
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-demo
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-charmed
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-ubuntu
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-maas
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-ceph
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-kubernetes
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=hpc-openstack
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=openinfra-summit
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=supercomputing
https://ubuntu-com-11956.demos.haus/hpc/contact-us?product=isc-hpc

## Issue / Card

Fixes https://github.com/canonical/web-squad/issues/5857

## Screenshots

image

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

### CODE REVIEW

File: templates/hpc/contact-us.html [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  returnURL="/openstack/thank-you?product=isc-hpc" %}
+  {% include "hpc/_hpc-contact-us-form.html" %}
+{% endwith %}
+
+  {% else %}
+
+{% with h1="Contact us about HPC & Supercomputing",
+  intro_text="Fill in your details below and a member of our data center sales team will be in touch.",
+  formid="4537",
+  returnURL="/openstack/thank-you?product=hpc-demo" %}
```
Comment (minkyngkm) [inline templates/hpc/contact-us.html:112]:
```suggestion
  returnURL="/openstack/thank-you" %}
```

File: templates/hpc/_hpc-contact-us-form.html [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          <fieldset class="u-no-margin--bottom">
+            <h3>About your company</h3>
+            <ul class="p-list">
+              <li class="p-list__item">
+                <label for="company">Company name:</label>
+                <input required id="company" name="company" maxlength="255" type="text" />
+              </li>
+              <li class="p-list__item">
+                <label for="industry">Industry:</label>
+                <input required id="industry" name="title" maxlength="255" type="text" />
```
Comment (minkyngkm) [inline templates/hpc/_hpc-contact-us-form.html:50]: According to copy doc, industry is not required
```suggestion
                <input id="industry" name="title" maxlength="255" type="text" />
```

File: templates/hpc/_hpc-contact-us-form.html [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                <label for="lastName">Last name:</label>
+                <input required id="lastName" name="lastName" maxlength="255" type="text"/>
+              </li>
+              <li class="p-list__item">
+                <label for="email">Email address:</label>
+                <input required id="email" name="email" maxlength="255" type="email" pattern="^[^ ]+@[^ ]+\.[a-z]{2,26}$" />
+              </li>
+              <li class="p-list__item">
+                <label for="phone">Phone number:</label>
+                <input required id="phone" name="phone" maxlength="255" type="tel"/>
```
Comment (minkyngkm) [inline templates/hpc/_hpc-contact-us-form.html:26]: According to copy doc, phone number is not required
```suggestion
                <input required id="phone" name="phone" maxlength="255" type="tel"/>
```

File: templates/hpc/_hpc-contact-us-form.html [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                <input required id="industry" name="title" maxlength="255" type="text" />
+              </li>
+            </ul>
+          </fieldset>
+          <fieldset class="u-no-margin--bottom">
+            <h3>Your comments</h3>
+            <ul class="p-list">
+              <li class="p-list__item">
+                <label for="Comments_from_lead__c">What would you like to talk to us about?</label>
+                <textarea required id="Comments_from_lead__c" name="Comments_from_lead__c" rows="4" maxlength="2000"></textarea>
```
Comment (minkyngkm) [inline templates/hpc/_hpc-contact-us-form.html:59]: According to copy doc, this textarea is not required

```suggestion
                <textarea id="Comments_from_lead__c" name="Comments_from_lead__c" rows="4" maxlength="2000"></textarea>
```

File: templates/hpc/contact-us.html [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+{% with h1="Contact us about OpenStack & HPC",
+  intro_text="Fill in your details below and a member of our data center sales team will be in touch.",
+  formid="4537",
+  returnURL="/openstack/thank-you?product=hpc-openstack" %}
+  {% include "hpc/_hpc-contact-us-form.html" %}
+{% endwith %}
+
+{% elif product == 'hpc-isv' %}
+
+{% with h1="As a HPC & Supercomputing ISV let’s discuss partnership",
```
Comment (minkyngkm) [inline templates/hpc/contact-us.html:72]:
```suggestion
{% with h1="As a HPC & Supercomputing ISV let's discuss partnership",
```

File: templates/hpc/contact-us.html [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+{% with h1="Let’s talk at OpenInfra Summit",
+  intro_text="Fill in your details below and a member of our team will be in touch to arrange a meeting.",
+  formid="4537",
+  returnURL="/openstack/thank-you?product=openinfra-summit" %}
+  {% include "hpc/_hpc-contact-us-form.html" %}
+{% endwith %}
+
+{% elif product == 'supercomputing' %}
+
+{% with h1="Let’s talk at Supercomputing",
```
Comment (minkyngkm) [inline templates/hpc/contact-us.html:90]:
```suggestion
{% with h1="Let's talk at Supercomputing",
```

File: templates/hpc/contact-us.html [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  intro_text="Fill in your details below and a member of our team will be in touch to arrange a meeting.",
+  formid="4537",
+  returnURL="/openstack/thank-you?product=supercomputing" %}
+  {% include "hpc/_hpc-contact-us-form.html" %}
+{% endwith %}
+
+
+{% elif product == 'isc-hpc' %}
+
+{% with h1="Let’s talk at ISC HPC",
```
Comment (minkyngkm) [inline templates/hpc/contact-us.html:100]:
```suggestion
{% with h1="Let's talk at ISC HPC",
```

**Comment by minkyngkm** [comment]
LGTM @carkod but the form submission failed. I think the form needs to be update in marketo.
https://sentry.is.canonical.com/canonical/ubuntu-com/issues/28183/?query=is%3Aunresolved

image

File: templates/hpc/contact-us.html (hunk 1 above)
Comment (carkod) [reply templates/hpc/contact-us.html:112]: Oh wow, I didn't notice this in the QA, actually this should be `/hpc/thank-you`. Thanks

File: templates/hpc/contact-us.html [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  returnURL="/hpc/thank-you?product=isc-hpc" %}
+  {% include "hpc/_hpc-contact-us-form.html" %}
+{% endwith %}
+
+  {% else %}
+
+{% with h1="Contact us about HPC & Supercomputing",
+  intro_text="Fill in your details below and a member of our data center sales team will be in touch.",
+  formid="4537",
+  returnURL="/hpc/thank-you?product=hpc-demo" %}
```
Comment (minkyngkm) [inline templates/hpc/contact-us.html:112]:
```suggestion
  returnURL="/hpc/thank-you" %}
```

**Review by minkyngkm — APPROVED** [review]
QA done and LGMT!

image

Just one thing - line 112, the returnURL needs to be `/hpc/thank-you` I think.

**Pull request merged**